### PR TITLE
Job seeker showanddestroy #157

### PR DESCRIPTION
**Pull request by vijayakarumudi**
Closes AgileVentures/MetPlus_tracker#157
Jobseeker Show page with Delete and Cancel buttons.
Rspec and cucumber tests.

### CODE REVIEW

**Comment by vijayakarumudi** [comment]
Close this PR

**Comment by vijayakarumudi** [comment]
PR for the review.

File: app/views/job_seekers/show.html.haml [hunk 1]
```diff
@@ -21,6 +23,34 @@
...
+
+
+%br
+%br
+%div{class: 'row'}
+	
+	
+	.col-sm-2
+	= button_to 'Delete Jobseeker', job_seeker_path(@jobseeker), method: :delete,            | 
+      data: {confirm: "Delete Jobseeker #{@jobseeker.id}?"}, class: 'btn btn-danger'   |
```
Comment (patmbolger) [inline app/views/job_seekers/show.html.haml]: Recommend changing from #{@jobseeker.id} to #{@jobseeker.name} as the name of the person will be displayed on the *show* view but not the ID.

File: features/job_seeker.feature [hunk 2]
```diff
@@ -59,8 +59,20 @@ Scenario: edit Js Registration without password change
...
+Scenario: delete jobseeker 
+ Given I am on the JobSeeker Show page for "vijaya.karumudi@gmail.com"
+ Then I click and accept the "Delete Jobseeker" button
+ And I wait for 1 seconds
+ Then I should see "Jobseeker was deleted successfully."
+ 
+
+Scenario:cancel redirect to homepage
+ Given I am on the Jobseeker show page
+ Then I click link  "Cancel" button
```
Comment (patmbolger) [inline features/job_seeker.feature]: You can replace this with:

    I click the "Cancel" link

The "Cancel" is styled (via bootstrap) to look like a button, but is actually a link.

File: features/step_definitions/web_steps.rb [hunk 3]
```diff
@@ -92,6 +92,16 @@ def search_text text
   visit current_path
 end
 
+Then(/^I click link  "(.*?)" button$/) do |cancel|
+  %{I click link(cancel)}
+end
```
Comment (patmbolger) [inline features/step_definitions/web_steps.rb]: With the scenario step noted above you don't need this step.

**Comment by patmbolger** [comment]
Looks good overall but please see comments.

File: app/views/job_seekers/show.html.haml (hunk 1 above)
Comment (joaopapereira) [inline app/views/job_seekers/show.html.haml]: I agree

**Comment by joaopapereira** [comment]
Looks good, but please review the comments

File: app/views/job_seekers/show.html.haml (hunk 1 above)
Comment (vijayakarumudi) [inline app/views/job_seekers/show.html.haml]: I updated jobseeker.id to jobseeker. firstname.

File: features/job_seeker.feature (hunk 2 above)
Comment (vijayakarumudi) [inline features/job_seeker.feature]: I tried above step it is failed.

Comment (patmbolger) [inline features/job_seeker.feature]: Make sure you added the "Then" keyword, as in:

    Then I click the "Cancel" link

**Comment by vijayakarumudi** [comment]
I reviewed all the comments for this PR.

File: features/job_seeker.feature (hunk 2 above)
Comment (vijayakarumudi) [inline features/job_seeker.feature]: yes I added "Then" keyword

**Comment by joaopapereira** [comment]
Is there any reason for having multiple empty lines added to all the files?

File: features/job_seeker.feature [hunk 4]
```diff
@@ -59,8 +59,20 @@ Scenario: edit Js Registration without password change
...
+ Then I click and accept the "Delete Jobseeker" button
+ And I wait for 1 seconds
+ Then I should see "Jobseeker was deleted successfully."
+ 
+
+Scenario:cancel redirect to homepage
+ Given I am on the Jobseeker show page
+ Then I click link "Cancel" button
+ Then I should be on the home page
+
```
Comment (patmbolger) [inline features/job_seeker.feature]: There is no navigation step that corresponds to "Jobseeker show page".

You can fit this by using the same "JobSeeker Show page" as in the prior scenario.
Then the "cancel link" step can use the generic step.

In summary, I suggest changing the scenario steps to:
```
Given I am on the JobSeeker Show page for "vijaya.karumudi@gmail.com"
Then I click the "Cancel" link
Then I should be on the home page
```

File: features/step_definitions/web_steps.rb [hunk 5]
```diff
@@ -92,6 +92,18 @@ def search_text text
   visit current_path
 end
 
+Then(/^I click link "(.*?)" button$/) do |cancel|
+  %{I click link(cancel)}
+end
```
Comment (patmbolger) [inline features/step_definitions/web_steps.rb]: This step is not needed with the suggested scenario change noted above.

Comment (vijayakarumudi) [inline features/step_definitions/web_steps.rb]: I will remove the step after the changes.

File: features/job_seeker.feature (hunk 4 above)
Comment (vijayakarumudi) [inline features/job_seeker.feature]: I will update the scenario steps based on summary.

Comment (vijayakarumudi) [inline features/job_seeker.feature]: I updated above scenario. it is passed.

File: features/step_definitions/web_steps.rb (hunk 5 above)
Comment (vijayakarumudi) [inline features/step_definitions/web_steps.rb]: I removed the step.

**Comment by vijayakarumudi** [comment]
PR for review.

**Comment by patmbolger** [comment]
PR looks good to me.

**Comment by vijayakarumudi** [comment]
PR for review

**Comment by vijayakarumudi** [comment]
PR for review

**Comment by joaopapereira** [comment]
You are still changing a file just to add new lines. Please review that

File: features/step_definitions/web_steps.rb [hunk 6]
```diff
@@ -96,3 +96,10 @@ def search_text text
 
 
 
+
```
Comment (joaopapereira) [inline features/step_definitions/web_steps.rb]: This file do not need to be edited

**Comment by vijayakarumudi** [comment]
This is updated PR for review.

**Comment by joaopapereira** [comment]
Looks good going to merge

**Pull request merged**